lib/_stream_writable.js

```diff
@@ -473,8 +477,19 @@ function finishMaybe(stream, state) {
   if (need) {
     if (state.pendingcb === 0) {
       prefinish(stream, state);
-      state.finished = true;
-      stream.emit('finish');
+      stream.flushed = true;
+      if (stream.__flush) {
```

Comment:
I think I'd like this to always execute `__flush` – maybe add a default `__flush` to the prototype that just immediately calls `cb`.

Also, I've come around to @calvinmetcalf's proposed `_end` name. Mind renaming to that?